lib/topological_inventory/ansible_tower/receptor/collector.rb

```diff
@@ -53,12 +52,18 @@ def async_sweep_inventory(refresh_state_uuid, sweep_scope, total_parts, refresh_
         logger.sweeping(:finish, source, sweep_scope, refresh_state_uuid)
       end
 
+      def response_received!
```

Comment:
What is the "!" signifying in the method name?

I know in many Rails methods this is used to say that the method will raise an error if something fails, but it is not clear to me why it is used here.

Reply:
"!" is also used in ruby methods to pointing out that the method is changing internal state of object